2.train-VAE/utils/betatcvae.py

```diff
@@ -334,9 +334,10 @@ def tc_weights(model, subset_train_df, path=None):
     # Create a dataframe from the weight matrix
     weight_df = pd.DataFrame(weight_matrix)
 
-    # Save as parquet to use for heatmap
-    weight_df_dir = pathlib.Path("./results/tc_weight_matrix_encoder.parquet")
-    weight_df.to_parquet(weight_df_dir)
+    if path:
```

Comment:
Maybe something to consider when/if the times comes around for a refactor, but a lot of these changes are the same, except the name of the parquet file

i haven't looked deeply into the contents of the utils folder, but I imagine that much of these model-specific utils can be abstracted into a more versatile class. This way, any change you make to one utils, it will be automatically propagated to others (so you don't have to manually maintain consistency)

Reply:
I agree yes, there is definitely some redundancy that could be improved upon and I hope to do a PR in the future with that goal, but I think is probably out of scope for this one and would need to be its own PR!